src/library-authoring/components/LibraryComponents.tsx

```diff
@@ -65,28 +63,7 @@ const LibraryComponents = ({
     return result;
   }, [blockTypesData]);
 
-  const { showLoading, showContent } = useMemo(() => {
-    let resultShowLoading = false;
-    let resultShowContent = false;
-
-    if (isFetching && !isFetchingNextPage) {
-      // First load; show loading but not content.
-      resultShowLoading = true;
-      resultShowContent = false;
-    } else if (isFetchingNextPage) {
-      // Load next page; show content and loading.
-      resultShowLoading = true;
-      resultShowContent = true;
-    } else if (!isFetching && !isFetchingNextPage) {
-      // State without loads; show content.
-      resultShowLoading = false;
-      resultShowContent = true;
-    }
-    return {
-      showLoading: resultShowLoading,
-      showContent: resultShowContent,
-    };
-  }, [isFetching, isFetchingNextPage]);
+  const showLoading = isFetching || isFetchingNextPage;
```

Comment:
Removing the loading state when we already have data to make the transition smooth.
With loading | Without loading 
:---        |    :----   
![components-with-loading](https://github.com/open-craft/frontend-app-course-authoring/assets/849463/13e4943f-51e3-4c4a-a387-223a45c02471) | ![components-without-loading](https://github.com/open-craft/frontend-app-course-authoring/assets/849463/ef7d9fd9-37e1-45d4-9bad-d5e0c81ffee5)

Reply:
I think it looks much nicer without the loading, nice addition!